components/renku_data_services/project/apispec.py

```diff
@@ -12,13 +12,13 @@
 from renku_data_services.project.apispec_base import BaseAPISpec
 
 
-class KeyCloakId(RootModel[str]):
+class Keyword(RootModel[str]):
     root: str = Field(
         ...,
-        description="Member's KeyCloak ID",
-        example="123-keycloak-user-id-456",
+        description="A single keyword",
+        max_length=99,
         min_length=1,
-        pattern="^[A-Za-z0-9-]+$",
+        pattern="^[A-Za-z0-9\\s\\-_.]*$",
```

Comment:
Why do we have to validate the keywords? We can just allow any character, right? Or do we break the search if we allow any character?

Reply:
I thought it might be good to not allow any characters here. There's no limitation on the search side. I'll remove this constraint.

Reply:
Hey @m-alisafaee, I am sorry for going back and forth on this. I think it makes sense to make this more constrained as you originally had it. So lets keep it like you originally had it and relax it later if needed. Because we want the search to easily handle the keywords and not have to add logic for emojis or other random characters.